File: Data Structures & Algorithms/single-number/submission-0.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Integer> hashmap = new HashMap<>();
        for(int num: nums)
        {
            hashmap.put(num, hashmap.getOrDefault(num, 0) + 1);
        }
        for(int i = 0; i < nums.length; i++)
        {
            if(hashmap.get(nums[i]) == 1)
            {
                return nums[i];
            }
        }
        return 0;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        freq = [[] for i in range(len(nums) + 1)]

        for num in nums:
            hashmap[num] = hashmap.get(num, 0) + 1

        for i,cnt in hashmap.items():
            freq[cnt].append(i)

        res = []
        for i in range(len(freq) - 1, 0, -1):
            for x in freq[i]:
                res.append(x)

                if len(res) == k:
                    return res
        

                

            



        

        

        


        



        